sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -684,13 +684,23 @@ object GroupedAggregations {
             if (preceding.isEmpty) {
               windowOptionBuilder.unboundedPreceding()
             } else {
-              windowOptionBuilder.preceding(preceding.get)
+              if (orderType == DType.STRING) { // Bounded STRING bounds can only mean "CURRENT ROW".
+                windowOptionBuilder.currentRowPreceding()
+              }
+              else {
```

Comment:
nit: our style guidelines have the else on the same line as the previous }

Reply:
Thanks for catching that. Fixed.